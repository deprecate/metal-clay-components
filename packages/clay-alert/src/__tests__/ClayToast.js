import ClayToast from '../ClayToast';

let toast;

const spritemap = '../node_modules/lexicon-ux/build/images/icons/icons.svg';

describe('ClayToast', function() {
	afterEach(() => {
		if (toast) {
			toast.dispose();
		}
	});

	it('should render default markup', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			title: 'info',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render an toast with classes', function() {
		toast = new ClayToast({
			elementClasses: 'my-custom-class',
			message: 'message',
			spritemap: spritemap,
			title: 'info',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render an toast with id', function() {
		toast = new ClayToast({
			id: 'myId',
			message: 'message',
			spritemap: spritemap,
			title: 'info',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render a "danger" style toast', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			style: 'danger',
			title: 'Danger!',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render an "info" style toast', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			style: 'info',
			title: 'Info',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render a "success" style toast', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			style: 'success',
			title: 'Success!',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should render a "warning" style toast', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			style: 'warning',
			title: 'Warning!',
		});

		expect(toast).toMatchSnapshot();
	});

	it('should close a toast', function() {
		toast = new ClayToast({
			message: 'message',
			spritemap: spritemap,
			title: 'Success!',
		});

		toast.element.querySelector('button.close').click();

		expect(toast.visible_).toBeFalsy();
	});

	// eslint-disable-next-line
	it('should close automatically a toast after 5 seconds when message has no links', function() {
		jest.useFakeTimers();

		toast = new ClayToast({
			autoClose: true,
			message: 'message',
			spritemap: spritemap,
			title: 'Success!',
		});

		expect(setTimeout.mock.calls.length).toBe(1);
		expect(setTimeout.mock.calls[0][1]).toBe(5000);

		jest.runAllTimers();
		expect(toast.visible_).toBeFalsy();
	});

	// eslint-disable-next-line
	it('should close automatically a toast after 5 seconds when message has links', function() {
		jest.useFakeTimers();

		toast = new ClayToast({
			autoClose: true,
			message: 'message<a href="#">Link</a>',
			spritemap: spritemap,
			title: 'Success!',
		});

		expect(setTimeout.mock.calls.length).toBe(1);
		expect(setTimeout.mock.calls[0][1]).toBe(10000);

		jest.runAllTimers();
		expect(toast.visible_).toBeFalsy();
	});

	it('should close and destroy a toast', function() {
		toast = new ClayToast({
			destroyOnHide: true,
			message: 'message',
			spritemap: spritemap,
			title: 'Success!',
		});

		toast.element.querySelector('button.close').click();

		expect(toast.visible_).toBeFalsy();
		expect(toast.disposed_).toBeTruthy();
	});

	it('should fail when no message is passed', function() {
		expect(() => {
			toast = new ClayToast({
				spritemap: spritemap,
				title: 'title',
			});
		}).toThrow();
	});

	it('should fail when no spritemap is passed', function() {
		expect(() => {
			toast = new ClayToast({
				message: 'message',
				title: 'title',
			});
		}).toThrow();
	});

	it('should fail when no title is passed', function() {
		expect(() => {
			toast = new ClayToast({
				message: 'message',
				spritemap: spritemap,
			});
		}).toThrow();
	});
});
