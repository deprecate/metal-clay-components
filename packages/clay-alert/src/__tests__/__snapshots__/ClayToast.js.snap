// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayToast should render a "danger" style toast 1`] = `
<div class="alert alert-danger fade alert-dismissible show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-exclamation-full">
      <title>exclamation-full</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#exclamation-full"></use>
    </svg>
  </span>
  <strong class="lead">Danger!</strong>message</div>
`;

exports[`ClayToast should render a "success" style toast 1`] = `
<div class="alert alert-success fade alert-dismissible show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-check-circle-full">
      <title>check-circle-full</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#check-circle-full"></use>
    </svg>
  </span>
  <strong class="lead">Success!</strong>message</div>
`;

exports[`ClayToast should render a "warning" style toast 1`] = `
<div class="alert alert-warning fade alert-dismissible show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-warning-full">
      <title>warning-full</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#warning-full"></use>
    </svg>
  </span>
  <strong class="lead">Warning!</strong>message</div>
`;

exports[`ClayToast should render an "info" style toast 1`] = `
<div class="alert alert-info fade alert-dismissible show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-info-circle">
      <title>info-circle</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#info-circle"></use>
    </svg>
  </span>
  <strong class="lead">Info</strong>message</div>
`;

exports[`ClayToast should render an toast with classes 1`] = `
<div class="alert alert-info fade alert-dismissible my-custom-class show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-info-circle">
      <title>info-circle</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#info-circle"></use>
    </svg>
  </span>
  <strong class="lead">info</strong>message</div>
`;

exports[`ClayToast should render an toast with id 1`] = `
<div class="alert alert-info fade alert-dismissible show" id="myId" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-info-circle">
      <title>info-circle</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#info-circle"></use>
    </svg>
  </span>
  <strong class="lead">info</strong>message</div>
`;

exports[`ClayToast should render default markup 1`] = `
<div class="alert alert-info fade alert-dismissible show" role="alert">
  <button class="btn close btn-unstyled" aria-label="Close" type="button">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-times">
      <title>times</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#times"></use>
    </svg>
  </button>
  <span class="alert-indicator">
    <svg aria-hidden="true" class="lexicon-icon lexicon-icon-info-circle">
      <title>info-circle</title>
      <use xlink:href="../node_modules/lexicon-ux/build/images/icons/icons.svg#info-circle"></use>
    </svg>
  </span>
  <strong class="lead">info</strong>message</div>
`;
